ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -297,6 +367,19 @@ func (r *RayClusterReconciler) reconcilePods(instance *rayiov1alpha1.RayCluster)
 		if err := r.List(context.TODO(), &workerPods, client.InNamespace(instance.Namespace), filterLabels); err != nil {
 			return err
 		}
+
+		// delete the worker pod if it is marked unhealthy
+		for _, workerPod := range workerPods.Items {
```

Comment:
@DmitriGekhtman @sriram-anyscale Please help review this part. Trying to see if that works perfectly with autoscaling.

Reply:
Thank you for pointing this out.
The autoscaler does a health check of ray nodes itself, which we must flag off -- I'll take care of that.
As a bonus, that should provide a better separation of concerns and contribute to the overall stability of the system.
cc @wuisawesome

Reply:
I think as we design the health check. We do not disable the GCS health check for worker nodes but rely on it. @wilsonwang371 @iycheng